### ISSUE TITLE
Feature request: Minimise to tray

### ISSUE THREAD

**Issue by Gersonzao** [gamingdoom/datcord#46]

**Comment by andersrh** [comment]
this would be a great feature. a really important one i would say

**Comment by gamingdoom** [comment]
I totally agree about this being a needed feature, as having Datcord open all of the time to receive notifications isn't ideal. However, I am not really sure about how I would even go about doing this as it would probably require extensive modifications to Firefox's source code. Perhaps I could have it so that Datcord launches in headless mode when you close the window and re-opens normal mode when you open it again.

**Comment by PaganDegree** [comment]
Just wanted to add bump support to this request

**Comment by gamingdoom** [comment]
I have a prototype of this working currently. I have a cpp program to toggle between headless and non headless mode so that when you close datcord, headless mode launches and when you launch datcord again, it reopens. Notifications also work in this mode. This solution isnt really ideal because of the memory usage, as it is kind of the same as having datcord open all the time.

**Comment by andersrh** [comment]
well i think it's better than nothing. I have lots of RAM so my PC can handle it, but for people with low RAM, it might be an issue.

**Comment by gamingdoom** [comment]
Datcord now runs in the background in 0.5.0 which means that you receive notifications in the background. However there are still some things I would like to improve. First of all, I use the method described earlier which means that it uses extra ram.  Secondly, the implementation can be a little jank and lead to datcord not launching the first time if it exited improperly. Third, clicking on notifications currently does nothing so I would like to fix that. Finally, the app doesn't actually minimize to tray so I feel like I should have that happen in the next iteration.

**Comment by andersrh** [comment]
> Datcord now runs in the background in 0.5.0 which means that you receive notifications in the background. However there are still some things I would like to improve. First of all, I use the method described earlier which means that it uses extra ram. Secondly, the implementation can be a little jank and lead to datcord not launching the first time if it exited improperly. Third, clicking on notifications currently does nothing so I would like to fix that. Finally, the app doesn't actually minimize to tray so I feel like I should have that happen in the next iteration.

amazing. thanks for the great work!

**Comment by soulstenance** [comment]
> Datcord now runs in the background in 0.5.0 which means that you receive notifications in the background. However there are still some things I would like to improve. First of all, I use the method described earlier which means that it uses extra ram. Secondly, the implementation can be a little jank and lead to datcord not launching the first time if it exited improperly. Third, clicking on notifications currently does nothing so I would like to fix that. Finally, the app doesn't actually minimize to tray so I feel like I should have that happen in the next iteration.

Great work! I love any implementation of software that is libre and I'll take any chance to stick it to Google and their browser monopoly. Unfortunately, even though this does seem to run in the background, I have no way to call the app again after closing it, without a tray icon. When I open it from my system's app menu, it just says that it's already running and errors out. So I can never close it, which isn't ideal.

**Comment by gamingdoom** [comment]
I have implemented tray support. I just need to make the detection for if you launch it again to re-open the window a little better and this will be done.

**Comment by poltpolt** [comment]
also waiting for that feature, great work btw

**Comment by Gersonzao** [comment]
I'm no longer using dis(service)cord, but I guess I'm leaving this open because other people want it.

**Comment by gamingdoom** [comment]
The improvements are now available in 0.5.2.

**Comment by Ambient-Impact** [comment]
Awesome. Looks like the Flatpak is yet to be updated but I guess that might be out of your hands?

**Comment by gamingdoom** [comment]
Yep, still working on updating flatpak.

**Comment by Ambient-Impact** [comment]
No worries. Thanks!
